src/hotspot/share/classfile/javaClasses.inline.hpp

```diff
@@ -123,7 +123,7 @@ int java_lang_String::length(oop java_string) {
   return length(java_string, value);
 }
 
-bool java_lang_String::is_instance_inlined(oop obj) {
+bool java_lang_String::is_instance(oop obj) {
```

Comment:
I don't really see anything wrong with this.  is_instance() is called above in asserts but that shouldn't affect performance by increasing the size of generated code.

Reply:
`is_instance()` is called in non-assertion code as well, e.g. `ConstantPool::add_dumped_interned_strings`. By merging the two versions, one wouldn't use the potentially slower version by accident, and it's also more consistent with `is_instance` of other classes (e.g. `java_lang_Class`) in the same file.

Reply:
While `ConstantPool::add_dumped_interned_strings` might be an example of something that's not performance sensitive (even with dynamic CDS), the 0.02%/4Kb static size increase is less than concerning. Thanks for checking!